protoclass/detection/extract_edge_volume.m

```
@@ -0,0 +1,61 @@
+function [edge_vol ] = extract_edge_volume( in_vol, Method, Threshold)
+% EXTRACT_EDGE_VOLUME Function to extract edges from spatial blocks within the images of the
+% volume considering different levels of pyramid
+%     [ feature_mat_vol ] = extract_edge( in_vol,
+%     pyr_num_lev, Method, Threshold) 
+%
+% Required arguments:
+%     in_vol : 3D array
+%         Entire volume.
+%     Method : string
+%         The name of edge detection method {'canny', 'sobel', ....}.
+%
+%     Threshold: int 
+%         default = 0 , if mentioned this value will be used for canny edge
+%         detector. 
+% Return:
+%     feature_vol :  3D array 
+%         Entire volume 
+%
+
+    % Check if the threshold is specified or not 
+    if nargin <= 2
+        Threshold = 0 ;
+    elseif nargin == 1 
+        if exits ('in_vol')
+            Method = 'canny' ; 
+        else
+            error ('not enough input'); 
```

You need to raise the error in that way:


```
error('function_name:type_error', 'this is is an error');
```

example:

```
error('extract_hog_volume:IncorrectNumPyr', ['The level in the pyramid cannot be 0 or less.']);
```